web/src/modules/i18n/containers/I18nProvider.js

```diff
@@ -47,37 +47,54 @@ export class I18nProvider extends React.Component<PropsType, StateType> {
       interpolation: {
         escapeValue: false // Escaping is not needed for react apps: https://github.com/i18next/react-i18next/issues/277
       },
-      debug: buildConfig().featureFlags.developerFriendly
+      debug: buildConfig().featureFlags.developerFriendly,
+      detection: {
+        order: ['localStorage', 'navigator']
+      }
     })
+
     this.state = {
-      language: DEFAULT_LANGUAGE,
+      language: this.i18n.language,
       fonts: I18nProvider.getSelectedFonts(DEFAULT_LANGUAGE),
       i18nLoaded: true
     }
   }
 
   componentDidMount () {
-    this.setLanguage(this.props.language)
+    const language = this.props.language
+    if (language) {
+      this.updateLanguage(language)
+    }
   }
 
-  setLanguage (language: ?string) {
-    const targetLanguage = language || this.i18n.languages[0]
+  updateLanguage (targetLanguage: string) {
+    if (targetLanguage === this.i18n.languages[0]) {
+      this.updateLanguageState(targetLanguage)
+      return
+    }
 
     // Set i18next language to apps language
     this.i18n.changeLanguage(targetLanguage, () => {
-      const fonts = I18nProvider.getSelectedFonts(targetLanguage)
-      this.setState(prevState => ({ ...prevState, language: targetLanguage, fonts }))
-
-      this.props.setUiDirection(RTL_LANGUAGES.includes(targetLanguage) ? 'rtl' : 'ltr')
-      if (document.documentElement) {
-        document.documentElement.lang = targetLanguage
-      }
+      this.updateLanguageState(targetLanguage)
     })
   }
 
+  updateLanguageState = (targetLanguage: string) => {
+    const fonts = I18nProvider.getSelectedFonts(targetLanguage)
+    this.setState(prevState => ({ ...prevState, language: targetLanguage, fonts }))
+
+    this.props.setUiDirection(RTL_LANGUAGES.includes(targetLanguage) ? 'rtl' : 'ltr')
```

Comment:
Is `RTL_LANGUAGES` replaced in a different PR?

Reply:
Yes